operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakRealmImportJob.java

```diff
@@ -0,0 +1,213 @@
+                .withContainers(keycloakContainer)
+                .addToVolumes(secretVolume)
+                .withRestartPolicy("Never")
+                .endSpec()
+                .endTemplate()
+                .endSpec()
+                .build();
+    }
+
+    private Volume buildSecretVolume() {
+        return new VolumeBuilder()
+                .withName(volumeName)
+                .withSecret(new SecretVolumeSourceBuilder()
+                        .withSecretName(secretName)
+                        .build())
+                .build();
+    }
+
+    private Job createImportJob() {
+        var keycloakContainer = buildKeycloakJobContainer();
+        var secretVolume = buildSecretVolume();
+        var importJob = buildJob(keycloakContainer, secretVolume);
+
+        return importJob;
+    }
+
+    private Container buildKeycloakJobContainer() {
+        var keycloakContainer =
+            this
+                .existingDeployment
+                .getSpec()
+                .getTemplate()
+                .getSpec()
+                .getContainers()
+                .get(0);
+
+        var importMntPath = "/mnt/realm-import/";
+
+        var command = List.of("/bin/bash");
+
+        var override = "--override=false";
+
+        var commandArgs = List.of("-c",
+                "/opt/keycloak/bin/kc.sh build && " +
+                    "/opt/keycloak/bin/kc.sh import --file='" + importMntPath + getRealmName() + "-realm.json' " + override);
+
+        keycloakContainer
+                .setCommand(command);
+        keycloakContainer
+                .setArgs(commandArgs);
+        var volumeMounts = List.of(
+            new VolumeMountBuilder()
+                    .withName(volumeName)
+                    .withReadOnly(true)
+                    .withMountPath(importMntPath)
+                    .build());
+
+        keycloakContainer.setVolumeMounts(volumeMounts);
+
+        // Disable probes since we are not really starting the server
+        keycloakContainer.setReadinessProbe(null);
+        keycloakContainer.setLivenessProbe(null);
+
+        return keycloakContainer;
+    }
+
+
+    public void updateStatus(KeycloakRealmImportStatusBuilder status) {
+        if (existingDeployment == null) {
+            status.addNotReadyMessage("No existing Deployment found, waiting for it to be created");
+            return;
+        }
+
+        if (existingJob == null) {
+            Log.info("Job about to start");
+            status.addStartedMessage("Import Job will start soon");
+        } else {
+            Log.info("Job already executed - not recreating");
+            var oldStatus = existingJob.getStatus();
+            var lastReportedStatus = realmCR.getStatus();
+
+            if (oldStatus == null) {
+                Log.info("Job started");
+                status.addStartedMessage("Import Job started");
+            } else if (oldStatus.getSucceeded() != null && oldStatus.getSucceeded() >= 0) {
```

Comment:
shouldnt it be > 0 ?

Reply:
yup, and the following `getFailed`! Thanks! 👍

Reply:
Hmm, the tests should've caught this. :\ We'll need to improve the coverage.